PURL-TYPES.rst

```diff
@@ -253,6 +253,24 @@ github
       pkg:github/package-url/purl-spec@244fd47e07d1004
       pkg:github/package-url/purl-spec@244fd47e07d1004#everybody/loves/dogs
 
+githubactions
+------
+``githubactions`` for GitHub Actions:
+
+- The ``namespace`` is the user or organization. It is not case sensitive and
+  must be lowercased.
+- The ``name`` is the repository name. It is not case sensitive and must be
+  lowercased.
+- The ``subpath`` is used to point to the location of an action within a repository in the event there are multiple defined. It is not case sensitive and must be lowercased. If omitted, the package-url refers to the default action for the repository only.
+- The ``version`` is a commit SHA or release tag. 
+- The ``repository_url`` refers to the GitHub instance that hosts the action you are using. If omitted, the default is assumed to be `https://github.com/`.
+- Examples::
+
+      pkg:githubactions/package-url/purl-spec@244fd47e07d1004
+      pkg:githubactions/package-url/purl-spec@v1.2
+      pkg:githubactions/github/codeql-action/analyze@v2
```

Comment:
It appears like the name attribute could be a single word `purl-spec` or could have subpath `codeql-action/analyze`. This appear to be different from other schemes where subpath are after qualifiers separated by a hash.

```
pkg:githubactions/github/codeql-action@v2?repository_url=...#subpath=analyze
```

Reply:
I think subpath has a special meaning for referring to files within a package, which is not the case here if the action is its own package which happens to be in the same repository as other packages. This seems consistent with Go where the namespace/name is treated as a single value which means something in Go instead of two separate values. (eg a lot of Go PURLs have the PURL name "v2" because of the way Go handles version epochs)

Reply:
@matt-phylum has it. The subpath is about picking a non-default action in a repository in instances where multiple actions exist.

Reply:
This seems inconsistent with the description given above:

> ``namespace`` is the user or organization
> ``name`` is the repository name
> ``subpath`` is used to point to the location of an action within a repository in the event there are multiple defined.

If the `name` component is strictly the repository name, then based on the components from the spec:

`scheme:type/namespace/name@version?qualifiers#subpath`

This example should be:

`pkg:githubactions/github/codeql-action@v2#analyze`